_search-plugins/knn/filter-search-knn.md

```diff
@@ -0,0 +1,649 @@
+                    }
+                }
+            }
+        }
+    }
+}
+```
+#### Sample response
+
+Upon success, you should receive "200-OK" status with the following response:
+
+```json
+{
+  "acknowledged" : true,
+  "shards_acknowledged" : true,
+  "index" : "hotels-index"
+}
+```
+
+## Step 2: Add data to your index
+
+Next, add data to your index with a PUT HTTP request. Make sure that the search criteria is defined in the body of the request.
+
+#### Sample request
+
+The following request adds twelve hotel documents that contain criteria such as feedback ratings and whether or not parking is available.  
+
+```json
+POST /_bulk
+{ "index": { "_index": "hotels-index", "_id": "1" } }
+{ "location": [5.2, 4.4], "parking" : "true", "rating" : 5 }
+{ "index": { "_index": "hotels-index", "_id": "2" } }
+{ "location": [5.2, 3.9], "parking" : "false", "rating" : 4 }
+{ "index": { "_index": "hotels-index", "_id": "3" } }
+{ "location": [4.9, 3.4], "parking" : "true", "rating" : 9 }
+{ "index": { "_index": "hotels-index", "_id": "4" } }
+{ "location": [4.2, 4.6], "parking" : "false", "rating" : 6}
+{ "index": { "_index": "hotels-index", "_id": "5" } }
+{ "location": [3.3, 4.5], "parking" : "true", "rating" : 8 }
+{ "index": { "_index": "hotels-index", "_id": "6" } }
+{ "location": [6.4, 3.4], "parking" : "true", "rating" : 9 }
+{ "index": { "_index": "hotels-index", "_id": "7" } }
+{ "location": [4.2, 6.2], "parking" : "true", "rating" : 5 }
+{ "index": { "_index": "hotels-index", "_id": "8" } }
+{ "location": [2.4, 4.0], "parking" : "true", "rating" : 8 }
+{ "index": { "_index": "hotels-index", "_id": "9" } }
+{ "location": [1.4, 3.2], "parking" : "false", "rating" : 5 }
+{ "index": { "_index": "hotels-index", "_id": "10" } }
+{ "location": [7.0, 9.9], "parking" : "true", "rating" : 9 }
+{ "index": { "_index": "hotels-index", "_id": "11" } }
+{ "location": [3.0, 2.3], "parking" : "false", "rating" : 6 }
+{ "index": { "_index": "hotels-index", "_id": "12" } }
+{ "location": [5.0, 1.0], "parking" : "true", "rating" : 3 }
+```
+
+#### Sample response
+
+Upon success, you should receive "200-OK" status with entries for each document ID added to the index. The following response is truncated to only show one document:
+
+```json
+{
+  "took" : 140,
+  "errors" : false,
+  "items" : [
+    {
+      "index" : {
+        "_index" : "hotels-index",
+        "_id" : "1",
+        "_version" : 2,
+        "result" : "updated",
+        "_shards" : {
+          "total" : 1,
+          "successful" : 1,
+          "failed" : 0
+        },
+        "_seq_no" : 12,
+        "_primary_term" : 3,
+        "status" : 200
+      }
+    }
+  ]
+}
+
+```
+
+## Step 3: Search your data with a filter
+
+Now you can create a k-NN search that specifies filters using Query DSL Boolean clauses. You need to include your reference point to search for nearest neighbors. Provide an x-y coordinate for the point with the `vector` field, such as `"vector": [ 5.0, 4.0]`
```

Comment:
Instead of just "using", do we mean "by using"? Otherwise, this can be read as though the filters are using the clauses. Instead of "with", do we mean "within"? Also, it looks like we need a terminating period.